engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -1871,6 +1908,204 @@ public static long minus(@Nullable final ZonedDateTime dateTime1, @Nullable fina
+     * Subtracts two durations.
+     *
+     * @param duration1 first duration
+     * @param duration2 second duration
+     * @return {@code null} if either input is {@code null}; otherwise the difference of the two durations
+     */
+    public static Duration minus(@Nullable final Duration duration1, @Nullable final Duration duration2) {
+        if (duration1 == null || duration2 == null) {
+            return null;
+        }
+
+        try {
+            return duration1.minus(duration2);
+        } catch (Exception ex) {
+            throw new DateTimeOverflowException(ex);
+        }
+    }
+
+    /**
+     * Subtracts two periods.
+     *
+     * @param period1 first period
+     * @param period2 second period
+     * @return {@code null} if either input is {@code null}; otherwise the difference of the two periods
+     */
+    public static Period minus(@Nullable final Period period1, @Nullable final Period period2) {
+        if (period1 == null || period2 == null) {
+            return null;
+        }
+
+        try {
+            return period1.minus(period2);
+        } catch (Exception ex) {
+            throw new DateTimeOverflowException(ex);
+        }
+    }
+
+    /**
+     * Multiply a duration by a scalar.
+     *
+     * @param duration the duration to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the duration multiplied by the scalar
+     */
+    public static Duration multiply(final Duration duration, final long scalar) {
+        if (duration == null || scalar == NULL_LONG) {
+            return null;
+        }
+
+        return duration.multipliedBy(scalar);
+    }
+
+    /**
+     * Multiply a duration by a scalar.
+     *
+     * @param duration the duration to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the duration multiplied by the scalar
+     */
+    public static Duration multiply(final long scalar, final Duration duration) {
+        return multiply(duration, scalar);
+    }
+
+    /**
+     * Multiply a duration by a scalar.
+     *
+     * @param duration the duration to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the duration multiplied by the scalar
+     */
+    public static Duration multiply(final Duration duration, final double scalar) {
+        if (duration == null || scalar == NULL_DOUBLE) {
+            return null;
+        }
+
+        if (Double.isNaN(scalar)) {
+            throw new DateTimeOverflowException("Scalar value is NaN");
+        }
+
+        final double product = duration.toNanos() * scalar;
+
+        if (product > Long.MAX_VALUE || product < Long.MIN_VALUE) {
+            throw new DateTimeOverflowException("Product value is too large to be cast to a long");
+        }
+
+        return Duration.ofNanos((long) product);
```

Comment:
floating point multiplication is a bit more speculative IMO; now, we are applying an implicit rounding mode.

Reply:
@cpwright and @rbasralian don't want to support this because it is not in the Java time lib.  Removed.